### ISSUE TITLE
Calling callback JS functions in Nim

### ISSUE THREAD

**Issue by Ethosa** [openpeeps/denim#10]
At JS side I pass some functions in Nim procedure:
```javascript
...
lib.callFunc(() => {
  console.log(1);
});
...
```

But at Nim side I have error on `denim build`:

```shell
Error: internal error: getTypeDescAux(tyEmpty)
No stack traceback available
To create a stacktrace, rerun compilation with './koch temp c <file>', see https://nim-lang.github.io/Nim/intern.html#debugging-the-compiler for details
```

Nim code:
```nim
init proc(module: Module) =

  proc callFunc(callback: napi_function): void {.export_napi.} =
    discard callFunction(args.get("callback"), [])
```

Nim version:
```shell
Nim Compiler Version 2.0.0 [Linux: amd64]
Compiled at 2023-08-01
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: a488067a4130f029000be4550a0fb1b39e0e9e7c
active boot switches: -d:release
```
UPD: on 1.6.14 I got same error
UPD2: same on 1.6.8

**Comment by Ethosa** [comment]
Here is reason https://github.com/openpeeps/denim/blob/main/src/denimpkg/napi/bindings.nim#L438

right version:
```nim
proc callFunction*(fn: napi_value, args: openarray[napi_value] = [], this: napi_value = %*(emptyArr)): napi_value =
```

**Comment by Ethosa** [comment]
So ... I'll send PR in near future

**Comment by Ethosa** [comment]
solved in #12
